### DISCUSSION TITLE
Downloadhelper icon does not show anymore on firefox addon

### DISCUSSION THREAD

**Discussion by uhlstefa** [aclap-dev/video-downloadhelper#129] [Help and Q&A]
### Your question

Hi,
a few days ago downloadhelper disappeared from my FF browser. Even reinstalling it didn't help. The Icon is nowhere to be found. Any ideas what to do?

### Addon & CoApp version and details

downloadhelper: 8.2.0.23
firefox 1222.0.2

**Comment by jcfrog** [comment]
hi,
please read this: https://github.com/aclap-dev/video-downloadhelper/wiki/Lost-icon

**Comment by uhlstefa** [comment]
Thx @jcfrog I will check and report back

**Comment by TonyPatriarche** [comment]
Same problem, slight variation:  FireFox now compresses all or most addons into one icon, a jigsaw piece, which pulls down a menu of all installed addons, from which they can be executed. This is a bit laborious, so I tapped the settings wheel button  and selected the option "display icon on toolbar" or words to that effect. 
1) downloadhelper disappeared from the FF pull-down menu -- very confusing
2) what toolbar? where? what icon?
... FF (plus Windows) has several areas that could be considered toolbars. I was just looking in all the wrong ones ;-) I tried the link in the answer above, and immediately saw the icon, right where it (obviously) should be, in the same row as the jigsaw-piece icon for the addons menu. The attached image shows the toolbar as it appears in my FF, with downloadhelper second from right. 
![FireFox Toolbar](https://github.com/aclap-dev/video-downloadhelper/assets/91024754/e5dd80d2-4163-4d51-af5c-4c0ffbe124e2)
Problem solved. I hope this helps anyone who is confused as I was. However, there remains a problem, or at least a wish-list item:

* downloadhelper should not disappear from the pull-down menu when the settings wheel "icon on toolbar" is selected. That's counter-intuitive IMHO, and not like most other menus, e.g. in the Windows menu, if I select "pin to taskbar", the icon appears on my taskbar, but it is still available on the Windows menu too.

**Reply by jcfrog** [reply]
Hi,
Sorry I'm not sure to understand, the pinned icon does not remain in the toolbar?

**Comment by muesli-joe** [comment]
hello,
i don't know if it's appropriate to piggyback this question or if it would be better to start a new question...

i have the same problem:
the vdh-icon does not appear in the tool-bar
...when i look to pin it to the toolbar, it is not listed
![image](https://github.com/user-attachments/assets/e03adbff-83f4-406e-8d6d-0ffb4f8286e9)
when i click "manage extensions" it is listed as installed and enabled.
reinstalling does not help.

does anyone have an idea?
...unfortunately everything i found in the forum was not helpful in my case (i found #514 #428 #453 and thisone)

version of vdh: 9.1.0.44
version of co-app: i cannot find the version, since the guide i found [here](https://github.com/aclap-dev/video-downloadhelper/wiki/Where-can-I-find-my-versions%3F) needs me to click on the icon, which is not present. i have however just installed the latest version using the curl-command.
i'm using firefox 128.3.1esr on debian bullseye kernel 5.10.0-33-amd64 with cinnamon 4.8.6 

i would really appreciate it, if anyone can point me in the right direction.

**Comment by jcfrog** [comment]
did you try here? 
![toolbar](https://github.com/user-attachments/assets/f882b064-ed97-41d6-821e-99adb7b36c21)

**Reply by muesli-joe** [reply]
thanks for the fast answer
yes, it's neither in the overflow nor in the customizing-menue 

![image](https://github.com/user-attachments/assets/bc90dd96-ed4b-47c3-9238-8ba8d537c22e)
![image](https://github.com/user-attachments/assets/aa03bc28-9af2-44a6-b69a-493a23eea479)

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  (@jcfrog Jérôme does, though.)  I'm just another user.

That's really strange.  I've never encountered that in Windows.  Maybe there's something weird about Linux.  I wouldn't know, being totally ignorant of Linux.  But as you can see, I can spell it.  Fat lot of help that is.

Try just disabling then re-enabling VDH.  On my Windows Firefox, that causes the icon to disappear from the toolbar, then reappear.  Maybe it will work for you.  You might also search Bugzilla https://bugzilla.mozilla.org/ to see if there's been any reports of extensions not getting their icon on the toolbar.

You are running a somewhat out of date version of Firefox.  I've got 132.0.2 here in Windows 11.  Isn't something newer available for Linux?  If there is, it might be worth a shot.  On the other hand, I've never seen this problem even back when I was on Windows Firefox versions way older than yours & on Windows 7 so . . .

**Reply by GrampaWildWilly** [reply]
To find your version of the CoApp, first find the directory where it resides.  On Windows, the executable is named vdhcoapp.exe.  Execute that program, whatever it's named, like this:

vdhcoapp --version

Two dashes.  This is how it works on my system.  You'll have to figure it out for Linux.  I would assume it's pretty much the same.  That will tell you the version number.
